### PR TITLE
AgContext form and Metadata form with jsonforms

### LINKED ISSUE #115: Fix up specification of BBCH growth stage codes
1. Currently called "bbch_code" should be "bbch_growth_stage"
2. Currently specified with gs prefix. This should be made more explicit in the description or with examples, or the gs prefix should be removed.
3. How do we want to allow a range or several BBCH growth stages to be specified? The growth stage is intended to be applied to a single plant, not a plantation.

### LINKED ISSUE #146: Agcontext editor form
Research options for react form entry options and develop a tool to allow users to create and export an AgContext yaml based on what they have entered into the form.

### PR DESCRIPTION
**Pull request by jnothman**
Fixes #146 
Fixes #115
Supersedes and closes #151

### CODE REVIEW

**Comment by jnothman** [comment]
This is now based off #200 to separate out that feature (and fix tests here)

**Comment by jnothman** [comment]
Far from perfect, but acceptable for merge in place of rjsf. Unfortunate that I've not got the growth stage component working.

**Comment by jnothman** [comment]
I'm rebasing this off the AgContext-Form-Integration branch so that we can see them together.

**Comment by jnothman** [comment]
Okay, this now includes #193

File: search/package.json [hunk 1]
```diff
@@ -20,17 +27,13 @@
...
     "react-router-dom": "^5.2.0",
     "react-scripts": "3.4.3",
     "redux": "^2.0.0",
-    "rjsf-material-ui": "^0.3.9",
-    "typescript": "^2.8.0"
+    "typescript": "^4.1.3"
   },
   "scripts": {
-    "start": "PORT=80 react-scripts start",
+    "start": "HOST=localhost react-scripts start",
```
Comment (ElevnLi) [inline search/package.json:36]: we should keep PORT=80, otherwise development mode will be on port 3000 by default.

**Comment by jnothman** [comment]
Sorry I didn't mean to commit a port change.  I had a conflict between two things running locally

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: search/package.json [hunk 2]
```diff
@@ -4,15 +4,23 @@
...
     "@testing-library/user-event": "^7.1.2",
+    "@types/jest": "^26.0.20",
+    "@types/node": "^14.14.22",
+    "@types/react": "^17.0.0",
+    "@types/react-dom": "^17.0.0",
     "js-cookie": "^2.2.1",
+    "js-yaml": "^4.0.0",
     "node-fetch": "^2.6.1",
     "node-forge": "^0.10.0",
     "react": "^16.13.1",
```
Comment (jnothman) [inline search/package.json:26]: Somehow the react-jsonschema-form dependencies are still here, when they were removed in the initial PR. Was this due to an incorrect resolution of a merge conflict?

**Pull request merged**